packages/origin-ui-core/src/features/users/sagas.ts

```diff
@@ -91,7 +92,10 @@ function* fetchOffchainUserDetails(): SagaIterator {
 
             let organization: IOrganizationWithRelationsIds = null;
 
-            if (typeof userProfile.organization !== 'undefined') {
+            if (
+                typeof userProfile.organization !== 'undefined' &&
+                [Role.Admin || Role.SupportAgent].includes(userProfile.organization)
```

Comment:
@JGiter could you explain this line? I'm trying to understand what Roles have to do with thr organization ID.

Reply:
In this block the user organization is fetched and if user doesn't have necessary rights the error is thrown

Reply:
Why are you comparing Roles to the Organization ID? The organization ID has no connection to Roles.
Roles are checked from `user.rights`, nothing to do with the organization.

You can also use the `isRole()` [helper function](https://github.com/energywebfoundation/origin/blob/1753e04781a422782d58874e8456e81d411cd460/packages/origin-backend-core/src/User.ts#L47).

`isRole(userProfile, Role.Admin, Role.SupportAgent)`.

Reply:
You right. Its typo